src/generate/peripheral.rs

```diff
@@ -16,6 +18,31 @@ use anyhow::{anyhow, bail, Context, Result};
 
 use crate::generate::register;
 
+/// An enum defining how to represent registers in the rendered outputs, binding a bool
+/// which describes whether or not renderers should include type information, 
+/// which allows for disjoint arrays to share the same type.
+#[derive(Debug)]
+enum Repr {
+    ArrayAndInfo(bool),
+    ExpandAndInfo(bool),
+    // String to store the common type when the erc is part of a disjoint array
+    SingleAndInfo(bool, String),
+}
```

Comment:
I think this could do with named fields instead of a tuple variant

```suggestion
#[derive(Debug)]
enum Repr {
    Array {
    	include_info: bool,
    },
    Expand {
    	include_info: bool,
    },
    // String to store the common type when the erc is part of a disjoint array
    Single {
        common: String,
    	include_info: bool,
    },
}
```

Im not a fan of these names and I'm sure they can be improved

Reply:
Yes and yes, the suggestion is fine if it looks good to you